README.md

```diff
@@ -1 +1,278 @@
+   the name `new-pass` and `next`.  `next` is the URL that Goat Guardian will
+   forward the end-user to after changing their password.  It is used in the next
+   step.  It could be something like `http://localhost:3000`.
+
+1. When the end-user submits this form, Goat Guardian will change the password
+   of the end-user and forward them to the URL in the `next` parameter.
+
+### Other Endpoints
+
+There are a few other endpoints provided by Goat Guardian.  They are described below.
+
+#### Logout
+
+Goat Guardian provides an endpoint that lets the end-user logout.  The flow for using this is described below:
+
+1. Have the end-user send a request to `/logout` with a query parameter called `next`.  `next` will be the URL to redirect the end-user to after logging them out.
+
+1. Goat Guardian will intercept this request and delete the session cookie for
+   the end-user.  Goat Guardian will then redirect the end-user to the URL from
+   the `next` parameter.  Since the end-user has been logged out, Goat Guardian
+   will not set the `X-UserId` HTTP header.
+
+## Options
+
+This section lists all the options that can be specified to Goat Guardian.
+
+Command Line Flag | Environment Variable | Default | Description
+----------------- | -------------------- | ------- | -----------
+`--twitter-oauth-callback` | `GG_TWITTER_OAUTH_CALLBACK_URL` | `http://localhost:3000/twitter/callback` | The callback for the Twitter OAuth.  The user will be redirected here after logging in on Twitter.  This should be a URL controlled by Goat Guardian.
+`--twitter-oauth-key` | `GG_TWITTER_OAUTH_KEY` | *(no default)* | The OAuth key to use for Twitter.  This should be a value like `Egex7CUsqQqSoEDjMivGtT1r0`.  You'll need to register with Twitter to get this value.
+`--twitter-oauth-secret` | `GG_TWITTER_OAUTH_SECRET` | *(no default)* | The OAuth secret to use for Twitter.  This should be a value like `HCZPPaOmib64GP7QbQaEwrLwKswK8pQDe4UwsAS3EVJBupBj5l`.  You'll need to register with Twitter to get this value.
+`--db-conn-string` | `GG_DB_CONN_STRING` | `:memory:` | The connection string to use for the SQLite database used to store the end-user accounts.  By default it just uses an in-memory database which is deleted when Goat Guardian stops running.  You should probably specify this as something like `/some/path/to/my-db.sqlite3` to save the end-user accounts to a file on disk.
+`--session-key` | `GG_SESSION_KEY` | *(no default)* | The encryption key used to encrypt/decrypt the session cookie sent to the end-user.  This can be generated by running `goat-guardian --generate-session-key`.
+`--redir-after-login-url` | `GG_REDIR_AFTER_LOGIN_URL` | `http://localhost:3000` | The URL to redirect the end-user to after logging in with Goat Guardian.
+
+## Example App
+
+There is an example Servant application implementing all of the authentiation flows described above.  You can find it in the [example-servant-app](./example-servant-app/) directory.
+
```

Comment:
It would be better to add sections about "Future work" and "Competitors".

I guess future work contains current contents.
* providing other authentication methods such as FaceBook
* providing a way to link several authentication methods (e.g., it enables end-users to login *either* of email or twitter)
* would be used for micro services, which have multiple web application servers behind goat guardian

In "Competitor" section, we should refer about `login-with` and other similar software.

Reply:
Thanks for the in-depth review!  I'll fix this next week!